tests/test-dirs/stack-or-heap.t/closures.ml

```diff
@@ -24,11 +24,6 @@ let f =
       (* ^ *)
 ;;
 
-let f =
-  exclave_ function | x -> x
-               (* ^ *)
-;;
-
```

Comment:
Why do we delete this test?

Reply:
I think we should instead replace with
```ocaml
let f x =
  exclave_ function | y -> y
               (* ^ *)
;;
```
Which gives
```
  |  exclave_ function | y -> y
  |                  ^
  
  |  exclave_ function | y -> y
  |           ^^^^^^^^^^^^^^^^^
  
  "stack"
```
(which it seems better preserves the original intent of the test)

Reply:
Ah yes thanks for catching this, I meant to leave a comment on it. That test caused a type error, which affects the other tests in the file since we conditionally run `optimize_allocations` based on there being no errors. I'll make the suggested change.